src/Vulkan/Silk.NET.Vulkan/VulkanLibraryNameContainer.cs

```diff
@@ -14,7 +14,7 @@ internal class VulkanLibraryNameContainer : SearchPathContainer
         public override string Linux => "libvulkan.so.1";
 
         /// <inheritdoc />
-        public override string MacOS => "libMoltenVK.dylib";
```

Comment:
this is a possible breaking change for downstream projects, is there a reason for this?

Reply:
This lib definition is incorrect.

The MoltenVK library (dylib) is considered as an ICD (Installable Client Driver) by the Vulkan loader. The Vulkan loader dynamically loads ICDs at execution time. When using the Vulkan loader, ICDs are not linked directly to the application. Similarly, the Vulkan loader dynamically loads the layers at execution time. Layers are not linked directly to the application.
The MoltenVK library is a little unusual because it is built in such a way that an application can link to it directly and run MoltenVK (Vulkan subset) applications without using the loader and layers. But if you want to use the loader and layers, you don't link MoltenVK and the layers to your application. You link only the loader. The loader then dynamically loads the MoltenVK lib as an ICD and dynamically loads the layers if requested.

Reply:
in resume it is impossible to create a vkInstance on MacOs binding to MoltenVK directly. Did anyone test this before?

Reply:
I'm happy with this change, iirc Ryujinx had to do this anyway to get it working on macOS.

Reply:
This is wrong and break Ryujinx see https://github.com/Ryujinx/Ryujinx/issues/4689.

Reply:
> This is wrong and break Ryujinx see [Ryujinx/Ryujinx#4689](https://github.com/Ryujinx/Ryujinx/issues/4689).

Our loader can now try multiple names, feel free to PR in something to try both libnames

Reply:
> This is wrong and break Ryujinx see [Ryujinx/Ryujinx#4689](https://github.com/Ryujinx/Ryujinx/issues/4689).

I will re-check the documentation and re-test.

Reply:
Will do if I find the time for that.

I should emphasis that this MoltenVK is built as `libMoltenVK.dylib` and never `libvulkan.dylib".

I know that a vulkan loader here could have done the job, but this is something **that should have been reported in the changelog as it's a breaking change.**

Reply:
> Will do if I find the time for that.
> 
> I should emphasis that this MoltenVK is built as `libMoltenVK.dylib` and never `libvulkan.dylib`.
> 
> I know that a vulkan loader here could have done the job, but this is something **that should have been reported in the changelog as it's a breaking change.**

Not listing it on the changelog is completely our fault (there were a *lot* of changes in 2.17, we surely were to have missed some), we will try to not miss breaking changes like this in the future!

Reply:
> Will do if I find the time for that.
> 
> I should emphasis that this MoltenVK is built as `libMoltenVK.dylib` and never `libvulkan.dylib`.
> 
> I know that a vulkan loader here could have done the job, but this is something **that should have been reported in the changelog as it's a breaking change.**

Found the time today to knock this down, PR opened https://github.com/dotnet/Silk.NET/pull/1419